lib/datasource/rubygems/get.ts

```diff
@@ -50,23 +50,13 @@ export async function getDependency(
     return null;
   }
 
-  const versions = (await fetch(dependency, registry, VERSIONS_PATH)) || [];
-
-  let releases = versions.map(
-    ({
-      number: version,
-      platform: rubyPlatform,
-      created_at: releaseTimestamp,
-      rubygems_version: rubygemsVersion,
-      ruby_version: rubyVersion,
-    }) => ({
-      version,
-      rubyPlatform,
-      releaseTimestamp,
-      rubygemsVersion,
-      rubyVersion,
-    })
-  );
+  let versions = [];
+  let releases = [];
+  try {
+    versions = await fetch(dependency, registry, VERSIONS_PATH);
+  } catch (err) {
```

Comment:
We need to check what type of error it is, and only continue if it's the one we expect from Artifactory, e.g. 404. If instead we get a 401 or 5xx response or connection failed, we *don't* want to continue and instead throw an error like usual.

Reply:
done